### PR TITLE
Add gamepad support (#122)

### PR DESCRIPTION
**Pull request by skejeton**
This works only for Windows for now.

### CODE REVIEW

**Comment by skejeton** [comment]
@marekmaskarinec requesting review...

File: umka/input.um [hunk 1]
```diff
@@ -273,3 +305,92 @@ fn getMouseScroll*(): th.Vf2 {
 	umth_input_get_mouse_scroll(&x, &y)
 	return th.Vf2{x, y}
 }
+
+fn umth_input_gamepad_get_players*(out: ^[4]int)
+
+//~~fn players
+// Returns a list of gamepads that are currently connected.
+fn players*(): []Player {
```
Comment (marekmaskarinec) [inline umka/input.um:313]: I don't like the use of the name `Player`. Why not just use `Gamepad`?

File: umka/input.um [hunk 2]
```diff
@@ -143,6 +143,38 @@ type Key* = enum {
 }
 //~~
 
+//~~Gamepad button number
+type GenericGamepadButton* = enum {
```
Comment (marekmaskarinec) [inline umka/input.um:147]: I think we can omit `Generic` here for the sake of brevity.

**Review by marekmaskarinec — CHANGES_REQUESTED** [review]

**Review by marekmaskarinec — APPROVED** [review]

**Comment by marekmaskarinec** [comment]
@skejeton thank you for the work! I merged this into main, linux support is tbd. Please test if it works after the merge.

**Comment by skejeton** [comment]
Ah Linux support was expected, I might work on it as well since I have the physical controller but it's a bit more involved. The merge works.

**Pull request merged**